src/printer.rs

```diff
@@ -75,85 +93,175 @@ impl<'a> Printer<'a> {
+                "{} ",
+                decorations
+                    .iter()
+                    .map(|seg| seg.text.to_owned())
+                    .collect::<Vec<String>>()
+                    .join(" ")
+            )?;
+        }
+
+        // Grid border.
+        let border = if gutter_width > 0 && self.config.output_components.grid() {
+            self.line_border()
+        } else {
+            PrintSegment {
+                size: 0,
+                text: "".to_owned(),
+            }
+        };
+
+        cursor_max -= border.size;
+        write!(self.handle, "{}", border.text)?;
+
+        // Line contents.
+        if self.config.output_wrap == OutputWrap::None {
+            let true_color = self.config.true_color;
+            let colored_output = self.config.colored_output;
+
+            write!(
+                self.handle,
+                "{}",
+                regions
+                    .iter()
+                    .map(|&(style, text)| as_terminal_escaped(
+                        style,
+                        text,
+                        true_color,
+                        colored_output
+                    ))
+                    .collect::<Vec<_>>()
+                    .join("")
+            )?;
+        } else {
+            for &(style, text) in regions.iter() {
+                let mut chars = text.chars().filter(|c| *c != '\n');
+                let mut remaining = text.chars().filter(|c| *c != '\n').count();
+
+                while remaining > 0 {
+                    let available = cursor_max - cursor;
+
+                    // It fits.
+                    if remaining <= available {
+                        let text = chars.by_ref().take(remaining).collect::<String>();
+                        cursor += remaining;
+
+                        write!(
+                            self.handle,
+                            "{}",
+                            as_terminal_escaped(
+                                style,
+                                &*text,
+                                self.config.true_color,
+                                self.config.colored_output,
+                            )
+                        )?;
+                        break;
+                    }
 
+                    // It wraps.
+                    let text = chars.by_ref().take(available).collect::<String>();
+                    cursor = 0;
+                    remaining -= available;
+
+                    write!(
+                        self.handle,
+                        "{}\n{}{}",
+                        as_terminal_escaped(
+                            style,
+                            &*text,
+                            self.config.true_color,
+                            self.config.colored_output,
+                        ),
+                        " ".repeat(gutter_width),
+                        border.text.to_owned()
+                    )?;
+
+                    continue;
```

Comment:
Is this `continue` statement needed?

Reply:
It appears not. Removing it in the next commit.